hail/build.gradle

```diff
@@ -98,6 +98,40 @@ compileScala.dependsOn(nativeLib)
 test.dependsOn(nativeLibTest)
 clean.dependsOn(nativeLibClean)
 
+task testCppCodegen(type: Test) {
+    useTestNG {}
+    
+    
+    // Enable C++ codegen for tests
+    environment HAIL_ENABLE_CPP_CODEGEN: '1'
+
+    filter {
+        includeTestsMatching "is.hail.nativecode.NativeCode"
+        includeTestsMatching "is.hail.annotations.AnnotationsSuite.testReadWrite"
+        includeTestsMatching "is.hail.annotations.UnsafeSuite"
+        includeTestsMatching "is.hail.io.IndexSuite"
+        includeTestsMatching "is.hail.vds.JoinSuite"
```

Comment:
I'm not sure why @cseed flagged joins as a good thing to test for this, but I don't see anything in JoinSuite that would go through the decoder.

I might include some tests that shuffle RVDs in here, since shuffles also use the encoder/decoder. I know there's a test in TableIRSuite that explicitly causes a shuffle, but I'm not sure where else there would be tests.

Reply:
Oops, yes, my mistake, I really meant shuffle, not join.

Reply:
I'm happy to change that if you want to suggest names of specific suites/tests that would be useful.

Reply:
`TableIRSuite.testShuffleAndJoinDoesntMemoryLeak` probably works as a test. I think that would be fine unless/until we decide to test the shuffling more explicitly, which afaict we're not doing in Scala right now.